packages/common/cmdk/package.json

```diff
@@ -8,5 +8,8 @@
     "react": "18.2.0",
     "react-dom": "18.2.0"
   },
+  "dependencies": {
+    "@radix-ui/react-dialog": "^1.0.5"
+  },
```

Comment:
This was used in https://github.com/toeverything/AFFiNE/blob/9650a5a6a1d1c7a144e101ba270f29f605c7ff5a/packages/common/cmdk/src/index.tsx#L3 without having been specified.

Reply:
That's strange, why didn't lint detect this issue? Let me double check.

Reply:
I think this is working currently because @radix-ui/react-dialog is specified in two other packages in the same workspace: https://github.com/toeverything/AFFiNE/blob/7c71c5457ae4e4cc98d487220e3b4614cc6dfe28/packages/frontend/core/package.json#L46
https://github.com/toeverything/AFFiNE/blob/7c71c5457ae4e4cc98d487220e3b4614cc6dfe28/packages/frontend/component/package.json#L37

So yarn hoists the package up to the root node_modules folder, which makes it resolvable by this package as well, even though the package itself doesn't specify it as a dependency. The node resolution algorithm just keeps looking up node_modules folders recursively until it reaches `/`, and doesn't actually care if it's in package.json or not. This is known as a "phantom dependency", here's a good writeup on the problems that might cause: https://rushjs.io/pages/advanced/phantom_deps/

I'm not aware of a lint rule that guards against phantom deps, but pnpm does guard against it by default (by symlinking only the specified deps for each package), and both npm and yarn have options you can use to opt-in to similar behavior:

- For npm it's [`--install-strategy=linked`](https://docs.npmjs.com/cli/v9/commands/npm-install#install-strategy).
- For yarn it's [`nodeLinker: pnpm`](https://yarnpkg.com/configuration/yarnrc#nodeLinker).

Reply:
Oh and I forgot to mention, Reflame of course will also guard against phantom deps. That's how I discovered these. :)

Reply:
we have  [`no-extraneous-dependencies`](https://github.com/toeverything/AFFiNE/blob/canary/.eslintrc.js#L271) rule for phantom deps. But it seems like the cmdk related code has disabled this rule, I will open another issue to record this problem.

> Oh and I forgot to mention, Reflame of course will also guard against phantom deps. That's how I discovered these. :)

Great!